src/parcsr_mv/new_commpkg.c

```diff
@@ -750,3 +753,58 @@ hypre_FillResponseIJDetermineSendProcs(void       *p_recv_contact_buf,
 
    return hypre_error_flag;
 }
+
+/*--------------------------------------------------------------------
+ * hypre_ParCSRCreateCommGraph
+ *
+ * Create communication topology graph for MPI neighborhood
+ * collectives
+ *--------------------------------------------------------------------*/
+
+#ifdef HYPRE_USING_NODE_AWARE_MPI
+void
+hypre_ParCSRCreateCommGraph(HYPRE_BigInt first_col_diag,
+                            HYPRE_BigInt *col_map_offd,
+                            MPI_Comm comm,
+                            hypre_ParCSRCommPkg *comm_pkg) {
+   HYPRE_Int num_sends = hypre_ParCSRCommPkgNumSends(comm_pkg);
+   HYPRE_Int num_recvs = hypre_ParCSRCommPkgNumRecvs(comm_pkg);
+   HYPRE_Int *send_map_starts = hypre_ParCSRCommPkgSendMapStarts(comm_pkg);
+   HYPRE_Int *recv_vec_starts = hypre_ParCSRCommPkgRecvVecStarts(comm_pkg);
+   HYPRE_Int *send_map_elmts =  hypre_ParCSRCommPkgSendMapElmts(comm_pkg);
+
+   int *sendcounts = (int *)malloc(num_sends * sizeof(int));
```

Comment:
We prefer using `hypre_TAlloc`, see https://github.com/hypre-space/hypre/blob/36ab29ba0977294a741987c4f921fe02840dff59/src/utilities/memory.h#L143 for allocating memory. In addition, it's preferable to use `hypre_` datatypes internally. So, `hypre_int` here.